src/ansys/geometry/core/designer/body.py

```diff
@@ -740,6 +767,27 @@ def translate(
             )
         )
 
+    @protect_grpc
+    @check_input_types
+    @reset_tessellation_cache
+    def rotate(
+        self,
+        axis_origin: Point3D,
+        axis_direction: UnitVector3D,
+        angle: Union[Quantity, Angle, Real],
+    ) -> None:  # noqa: D102
+        angle = angle if isinstance(angle, Angle) else Angle(angle)
+        rotation_magnitude = angle.value.m_as(DEFAULT_UNITS.SERVER_ANGLE)
+        self._grpc_client.log.debug(f"Rotating body {self.id}.")
+        self._bodies_stub.Rotate(
+            RotateRequest(
+                id=self.id,
+                axis_origin=point3d_to_grpc_point(axis_origin),
+                axis_direction=unit_vector_to_grpc_direction(axis_direction),
+                angle=rotation_magnitude,
+            )
+        )
```

Comment:
Is there no response coming from the server side?

Reply:
A true/false message should be enough to see whether the body was effectively rotated I guess.

Reply:
@RobPasMue `rotate()` is basically identical to `translate()` in function, so I wanted to stay consistent between the 2 in design